livekit-agents/livekit/agents/utils/codecs/mp3.py

```diff
@@ -39,6 +39,20 @@ def __init__(self):
         self._codec = av.CodecContext.create("mp3", "r")  # noqa
 
     def decode_chunk(self, chunk: bytes) -> List[rtc.AudioFrame]:
+        # Skip ID3v2 header if present
+        if chunk.startswith(b"ID3"):
+            # ID3v2 header is 10 bytes long
+            # The size is encoded in the next 4 bytes (bytes 6-9)
+            # Each byte only uses 7 bits (most significant bit is always 0)
+            if len(chunk) >= 10:
+                size = (
+                    ((chunk[6] & 0x7F) << 21)
+                    | ((chunk[7] & 0x7F) << 14)
+                    | ((chunk[8] & 0x7F) << 7)
+                    | (chunk[9] & 0x7F)
+                )
+                chunk = chunk[10 + size :]
```

Comment:
Weird that avcodec doesn't ignore it

Reply:
this is part of the container, usually decoders don't handle container-specific logic. libavformat handles it.. but currently the mp3 abstraction doesn't work well with it.

the "right" way of using it is to construct a stream each time we receive a new file, and then read the rest of the decoded stream.

since the abstraction is incorrect and would be a bigger change, I opted for a simpler path since mp3 headers are fairly simple to handle anyways.